kagenti/ui/Home.py

```diff
@@ -94,5 +96,52 @@
         """
     )
 
+# --- Login Redirect ---
+st.session_state[ENABLE_AUTH_STRING] = False
+ENABLE_AUTH = os.environ.get('ENABLE_AUTH')
+if ENABLE_AUTH == "true":
+    st.session_state[ENABLE_AUTH_STRING] = True
+
+    st.markdown("---")
+    st.subheader("Login")
+
+    # Example
+    # CLIENT_ID="kagenti"
+    # CLIENT_SECRET="xFPc7EPVV..."
+    # AUTH_ENDPOINT="http://localhost:8080/realms/master/protocol/openid-connect/auth"
+    # TOKEN_ENDPOINT="http://localhost:8080/realms/master/protocol/openid-connect/token"
+    # REDIRECT_URI="http://localhost:8502/oauth2/callback"
+    # SCOPE="openid profile email"
+    CLIENT_ID = os.environ.get('CLIENT_ID', 'kagenti')
+    CLIENT_SECRET = os.environ.get('CLIENT_SECRET')
+    AUTH_ENDPOINT = os.environ.get('AUTH_ENDPOINT')
+    TOKEN_ENDPOINT = os.environ.get('TOKEN_ENDPOINT')
+    REDIRECT_URI = os.environ.get('REDIRECT_URI')
+    SCOPE = os.environ.get('SCOPE', 'openid profile email')
+
+    oauth2 = OAuth2Component(CLIENT_ID, CLIENT_SECRET, AUTH_ENDPOINT, TOKEN_ENDPOINT)
+
+    # Check if token exists in session state
+    if TOKEN_STRING in st.session_state:
+        # User is not logged in
+        st.info("User is logged in")
```

Comment:
Please show something from the decoded token, such as user name or id, in this message.